streams/src/main/java/org/apache/kafka/streams/errors/ProductionExceptionHandler.java

```diff
@@ -34,6 +35,18 @@ public interface ProductionExceptionHandler extends Configurable {
     ProductionExceptionHandlerResponse handle(final ProducerRecord<byte[], byte[]> record,
                                               final Exception exception);
 
+    /**
+     * Handles serialization exception and determine if the process should continue. The default implementation is to
+     * fail the process.
+     *
+     * @param record        the record that failed to serialize
+     * @param exception     the exception that occurred during serialization
+     */
+    default ProductionExceptionHandlerResponse onSerializationException(final ProducerRecord record,
```

Comment:
The KIP say that we would add
```
handleSerializationException(ProducerRecord record, Exception exception); 
```

We need to stick to the KIP, or alter the KIP. But why would we need to alter the KIP?

Reply:
for the naming I'll change it back to the original kip's spec.  The only reason I would alter the kip is the exception type, it seems too loose as the name suggests it only handles the serialization exception.  But do you think it is necessary?

Reply:
I think it's correct to have `Exception` as it's more generic. Guess we could actually make it a `RuntimeException` but limiting it to `SerializationException` might be too restrictive, because the interface of `org.apache.kafka.common.serialization.Serializer#serialize` does not declare what exception might be throw, so it could be anything?